lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -129,8 +148,101 @@ data class GetTransactionStatusResponse(
 
     @SerialName("execution_status")
     val executionStatus: TransactionExecutionStatus? = null,
+
+    @SerialName("failure_reason")
+    val failureReason: String? = null,
 ) : StarknetResponse
 
+@Serializable
+data class GetMessagesStatusResponse(
+    @SerialName("transaction_hash")
+    val transactionHash: Felt,
+
+    @SerialName("finality_status")
+    val finalityStatus: TransactionStatus,
+
+    @SerialName("failure_reason")
+    val failureReason: String? = null,
+) : StarknetResponse
+
+@Serializable
+data class StorageProof(
+    @SerialName("classes_proof")
+    val classesProof: NodeHashToNodeMapping,
+
+    @SerialName("contracts_proof")
+    val contractsProof: ContractsProof,
+
+    @SerialName("contracts_storage_proofs")
+    val contractsStorageProofs: List<NodeHashToNodeMapping>,
+
+    @SerialName("global_roots")
+    val globalRoots: GlobalRoots,
+) : StarknetResponse
+
+@Serializable
+data class GlobalRoots(
+    @SerialName("contracts_tree_root")
+    val contractsTreeRoot: Felt,
+
+    @SerialName("classes_tree_root")
+    val classesTreeRoot: Felt,
+
+    @SerialName("block_hash")
+    val blockHash: Felt,
+)
+
+@Serializable
+data class MerkleNode(
+    @SerialName("path")
+    val path: Int,
+
+    @SerialName("length")
+    val length: Int,
+
+    @SerialName("value")
+    val value: Felt,
+
+    @SerialName("children_hashes")
+    val childrenHashes: ChildrenHashes? = null,
+)
+
+@Serializable
+data class ChildrenHashes(
+    @SerialName("left")
+    val left: Felt,
+
+    @SerialName("right")
+    val right: Felt,
+)
```

Comment:
I think we should call it `MerkleNodeChildrenHashes`, or even better make it a nested class within `MerkleNode`.